feature-libs/cart/base/components/added-to-cart-toast/added-to-cart-toast.component.spec.ts

```diff
@@ -0,0 +1,242 @@
+  OrderEntry,
+} from '../../root';
+import { AddedToCartToastComponentService } from './added-to-cart-toast-component.service';
+import { AddedToCartToastComponent } from './added-to-cart-toast.component';
+
+class MockWindowRef {
+  nativeWindow = window;
+  document = document;
+}
+const mockCartToastItem: CartToastItem = {
+  productName: 'Test Product',
+  quantity: 1,
+  baseClass: 'toast-container on-header close-toast',
+  unitPrice: '$3.45',
+  images: {},
+};
+const mockCartToastItems: CartToastItem[] = [mockCartToastItem];
+
+class MockAddedToCartToastComponentService
+  implements Partial<AddedToCartToastComponentService>
+{
+  addToast(
+    quantityAdded: number,
+    entries: OrderEntry,
+    containerClass: string
+  ): CartToastItem {
+    const { product } = entries;
+    const cartToastItem: CartToastItem = {
+      baseClass: containerClass,
+      productName: product?.name,
+      quantity: quantityAdded,
+      images: product?.images,
+      unitPrice: product?.price?.formattedValue,
+    };
+    mockCartToastItems.push(cartToastItem);
+    return cartToastItem;
+  }
+  getToasts(): Observable<CartToastItem[]> {
+    return of(mockCartToastItems);
+  }
+  setPosition(_className: string): void {}
+  removeToast(): void {}
+}
+
+const mockEvent = new CartUiEventAddToCart();
+mockEvent.quantity = 1;
+mockEvent.productCode = '';
+
+class MockActiveCartService implements Partial<ActiveCartFacade> {
+  getLastEntry(_productCode: string): Observable<OrderEntry | undefined> {
+    return of({});
+  }
+}
+
+describe('AddedToCartToastComponent', () => {
+  let component: AddedToCartToastComponent;
+  let fixture: ComponentFixture<AddedToCartToastComponent>;
+  let service: AddedToCartToastComponentService;
+  let activeCartService: ActiveCartFacade;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [I18nTestingModule, UrlTestingModule],
+      declarations: [AddedToCartToastComponent],
+      providers: [
+        { provide: WindowRef, useClass: MockWindowRef },
+        {
+          provide: AddedToCartToastComponentService,
+          useClass: MockAddedToCartToastComponentService,
+        },
+        {
+          provide: ActiveCartFacade,
+          useClass: MockActiveCartService,
+        },
+      ],
+    }).compileComponents();
+
+    service = TestBed.inject(AddedToCartToastComponentService);
+    activeCartService = TestBed.inject(ActiveCartFacade);
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AddedToCartToastComponent);
+    component = fixture.componentInstance;
+    component.timeout = 3000;
+    component.scrollEventUnlistener = () => {};
+    fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should call the active cart service', () => {
+    const spy = spyOn(activeCartService, 'getLastEntry').and.callThrough();
+    component.ngOnInit();
```

Comment:
why do we need ngOnInit() on every test case here? Could we achieve the same goal with beforeEach?